crates/matrix-sdk/src/widget_api/mod.rs

```diff
@@ -0,0 +1,16 @@
+//! Client widget API implementation.
+
+use crate::room::Room as JoinedRoom;
+
+pub mod permissions;
+pub mod widget;
```

Comment:
I think it's nice that the modules are internally kept small, but for the public API it's probably better (more convenient) if the types from these modules are re-exported here and the modules themselves are private.

```suggestion
mod permissions;
mod widget;

pub use self::{
    permissions::{Permissions, PermissionsProvider},
    widget::{Comm, Info, Widget},
};
```

Reply:
Agree 👍